test/afalgtest.c

```diff
@@ -24,26 +24,7 @@
 
 #ifndef OPENSSL_NO_ENGINE
 static ENGINE *e;
-#endif
-
 
-#ifndef OPENSSL_NO_AFALGENG
```

Comment:
I suppose this part should not be removed, or?

Reply:
this part is not necessary, so I removed it.
I tried with new kernel headers/old runtime kernel
and vice versa the test case is always passed.
ENGINE_by_id("afalg") != NULL should guarantee the engine is working.